tools/cldr-code/src/test/java/org/unicode/cldr/unittest/CheckLdmlDtdReferences.java

```diff
@@ -0,0 +1,342 @@
+        if (seen.contains(element) || SKIP_ELEMENTS.contains(element.getName())) {
+            return counter;
+        }
+        seen.add(element);
+
+        boolean elementShown = false;
+        if (missingElements.get(dtdType).contains(element)) {
+            counter = showMissing(dtdType, element, padding, false, counter);
+            elementShown = true;
+        }
+        Set<Attribute> attributes = element.getAttributes().keySet();
+        for (Attribute attribute : attributes) {
+            if (!attribute.isDeprecated()
+                && !SKIP_ATTRIBUTES.contains(attribute.getName())
+                && missingAttributes.get(dtdType).contains(attribute)) {
+                if (!elementShown) {
+                    counter = showMissing(dtdType, element, padding, true, counter);
+                    elementShown = true;
+                }
+                System.out.println(++counter + "\t" + dtdType + "\t" + padding + attribute.appendDtdString(new StringBuilder()));
+            }
+        }
+        Set<Element> children = element.getChildren().keySet();
+        for (Element child : children) {
+            if (!child.isDeprecated()) {
+                counter = showMissing(dtdType, child, padding + " ", seen, counter);
+            }
+        }
+        return counter;
+    }
+
+    public static int showMissing(DtdType dtdType, Element element, String padding, boolean ok, int counter) {
+        if (element.name.equals("script")) {
+            int debug = 0;
+        }
+        String message = padding + dtdType + "\t" + element.toDtdString();
+        if (ok) {
+            message += "\t🆗" + getBest(dtdType, element);
+        } else {
+            message += "\t❌" + getBest(dtdType, element);
+        }
+        System.out.println(++counter + "\t" + message);
+        return counter;
+    }
+
+    public static String getBest(DtdType dtdType, Element element) {
+        final Multimap<Element, String> elementToLink = foundElementsToLink.get(dtdType);
+        final Multimap<Element, Element> childToParents = childToParent.get(dtdType);
+        Element original = element;
+        String message = " ➡︎";
+        int i = 0;
+        while (true) {
+            Collection<String> links = elementToLink.get(element);
+            if (!links.isEmpty()) {
+                return message + " " + (i == 0 ? "" : element.getName()) + "\t" + Joiner.on('\t').join(links);
+            }
+            Collection<Element> parents = childToParents.get(element);
+            if (parents.isEmpty()) {
+                return message + "\tNONE";
+            }
+            element = parents.iterator().next(); // TODO fix to show all parents
+            message += "➡︎";
+            ++i;
+        }
+    }
+
+    private static void process(Path file) throws IOException {
+        DtdType dtdDtype = file.getFileName().toString().equals("info.md")
+            ? DtdType.supplementalData
+                : DtdType.ldml;
```

Comment:
- missing keyboard 
- push this logic into DtdType

Reply:
It really doesn't want to be there. This is a hack, just to get a better default; supplemental elements are all over.

Reply:
Perhaps we eventually want DTD annotations linking elements (just the top level, such as `localeDisplayNames`) to a section or anchor.

There's `DtdType.dtdPath` - this seems like a parallel to the dtdPath.